.github/workflows/_sonarcloud.yml

```
@@ -0,0 +1,90 @@
+name: Perform SonarCloud Scan
+
+on:
+  workflow_call: 
+
+    inputs:
+      sonar-config:
+        description: "Configuration for SonarCloud"
+        type: string
+        default: "default"
+      sonar-test-inclusions:
+        type: string
+        default: "/test"
+      sonar-exclusions:
+        type: string
+        default: "/test"
+      sonar-java-version:
+        type: number
+        default: 17
+      sonar-java-distribution:
+        type: string
+        default: temurin
+    secrets:
+      SONAR_TOKEN:
+        required: true
+
+permissions: {}
+  
+jobs:
+
+  quality:
+    name: Quality scan
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: read
+      pull-requests: write
+
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          fetch-depth: 0
+          ref: ${{  github.event.pull_request.head.sha }}
```

```suggestion
          ref: ${{ github.event.pull_request.head.sha }}
```
This extra space is all over the place!